src/rope/rope.rs

```diff
@@ -436,6 +436,27 @@ impl Rope {
         is_grapheme_boundary(self.chunks(), self.byte_len(), byte_offset)
     }
 
+    /// Returns true if this rope and `other` point to precisely the same
+    /// in-memory data.
+    ///
+    /// This happens when one of the ropes is a clone of the other and
+    /// neither have been modified since then.  Because clones initially
+    /// share all the same data, it can be useful to check if they still
+    /// point to precisely the same memory as a way of determining
+    /// whether they are both still unmodified.
+    ///
+    /// Note: this is distinct from checking for equality: two ropes can
+    /// have the same *contents* (equal) but be stored in different
+    /// memory locations (not instances).  Importantly, two clones that
+    /// post-cloning are modified identically will *not* be instances
+    /// anymore, even though they will have equal contents.
+    ///
```

Comment:
To be consistent w/ all the other public methods we should add an `# Examples` section to show in code how this works.